Unconditionally assign smarty-var, remove unused variables

eileenmcnaughton opened the pull request:
![image](https://user-images.githubusercontent.com/336308/231288895-7482dddf-ba4e-495b-9708-7bf7fce8f900.png)

eileenmcnaughton committed:
Unconditionally assign smarty-var, remove unused variables

demeritcowboy commented on CRM/Event/Form/ManageEvent/EventInfo.php:
```diff
@@ -33,18 +33,8 @@ public function preProcess() {
-    else {
-      $eventType = 'null';
-    }
+    $this->assign('entityID', $entityID);
```
This still gives an error because in the template it's entityId. Hmm...

demeritcowboy replied:
Having said that, it's mergeable since it's better and not worse.

eileenmcnaughton replied:
@demeritcowboy sigh - so should I just remove that line? is it not used If so I will remove because it would be not worse & there would be less useless code

demeritcowboy replied:
It's used but as entityId. I didn't check how far back it goes but it's possible it has never worked, so yes there's an argument to remove it AND the .hlp/.tpl code.

eileenmcnaughton commented:
@demeritcowboy I took another look at this & realised the never-worked `entityId` just needed an already-assigned eventID to work so I fixed it. I also 

- added `getEventID()` as a public method - I have been ensuring all event related forms have this. One form had already declared a protected `getEventId` which I removed in order to use the standard one.
- changes showHide to be a local variable rather than an undefined property
- assign `eventID` in the form rather than do a switcheroo in the template

I went through & all the Manage Event forms are loading notice free for me with full strict smarty + debugging

demeritcowboy commented:
Ok cool will take a look.

demeritcowboy commented:
I ran into https://lab.civicrm.org/dev/core/-/issues/4550 which makes it a little hard to test fully but I think this is good.

eileenmcnaughton commented:
Thanks @demeritcowboy & looks like you found a regression too...

colemanw commented:
Another regression: https://lab.civicrm.org/dev/core/-/issues/4706#note_152775
